core/jazz_splunk-kinesis-log-streamer/components/utils.js

```diff
@@ -21,11 +21,75 @@
 	@version: 1.0
 **/
 
+const AWS = require('aws-sdk');
+const Uuid = require("uuid/v4");
+const request = require('request');
 const logger = require("../components/logger.js");
 const global_config = require("../config/global-config.json");
 const truncate = require('unicode-byte-truncate');
 
 // Helper functions
+
+function assumeRole(configData, serviceData){
+  var isPrimary, roleArn;
+  if(serviceData){
+  	isPrimary = checkIsPrimary(serviceData.deployment_accounts[0].accountId, configData);
+  	roleArn = getRolePlatformService(serviceData.deployment_accounts[0].accountId, configData);
+  } 
```

Comment:
Should we handle the case when serviceData is null or undefined? Maybe log error and reject with error?

Reply:
`getServiceMetaData` should return error when it fails to retrieve service details.

Reply:
Done.